### PR TITLE
Add GitHub Actions CI

### PR DESCRIPTION
**Pull request by wojtekmaj**
Depends on #53 and #54, which fix the issues CI will now detect.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        with:
+          path: .yarn/cache
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.cache-name }}
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
```
Comment (NickGerleman) [inline .github/workflows/ci.yml:94]:
```suggestion
      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version: "18"
          cache: yarn
```

Comment (NickGerleman) [reply .github/workflows/ci.yml:94]: Same for above

Comment (wojtekmaj) [reply .github/workflows/ci.yml:94]: Last time I checked it didn't work correctly with modern Yarn, has anything changed?

Comment (NickGerleman) [reply .github/workflows/ci.yml:94]: Source code for it says probably
image

Comment (wojtekmaj) [reply .github/workflows/ci.yml:94]: Experimenting ➡️ https://github.com/rn-versions/rn-versions.github.io/pull/56

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      - main
+
+jobs:
+  lint:
+    name: Static code analysis
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (NickGerleman) [inline .github/workflows/ci.yml:15]: With the large history of this repo, one more optimization could be shallow checkout.

Comment (wojtekmaj) [reply .github/workflows/ci.yml:15]: Excellent point. In fact we should update all CI/CD workflows, not only the new ones.

**Pull request merged**